### PR TITLE
Test Cases for context.go

### PR DESCRIPTION
**Pull request by drakshakhan**
This PR adds test cases for the Cluster Context file, improving the test coverage from 0% to 75%. The following changes were made:

### **Test Case Addition** 
Added unit tests to thoroughly test the Cluster Context Command functionality.

### **Test Frameworks Used**
Testify for assertions and test suite management.
Interfaces for mocking methods and functions.
Assert for more expressive checks and validations.
Mock for creating mock implementations of interfaces.

### **Coverage Details**
The code coverage for the Cluster Context file has been increased to 75%. This includes tests for various scenarios, such as edge cases and failure paths.

### **Test Execution**
Tests were executed using the following commands:
go test -coverprofile=coverage.out ./... to run the tests and generate coverage information.
go tool cover -html=coverage.out to visualize the coverage and ensure completeness.

### **Verification**
All tests pass successfully, confirming that the added tests work as expected and improve the overall test coverage.

### CODE REVIEW

File: cmd/cluster/context_test.go [hunk 1]
```diff
@@ -0,0 +1,829 @@
...
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"bou.ke/monkey"
```
Comment (joshbranham) [inline cmd/cluster/context_test.go:14]: Can we just write interfaces where you need them instead of pulling in yet another mocking library/framework?

Comment (drakshakhan) [reply cmd/cluster/context_test.go:14]: Thanks for the feedback. I used the mocking framework as it required minimal changes. Since there are only a few interfaces, I’ll need to create additional ones to mock the behavior, which will involve more code changes. I’ll proceed with that approach as per your suggestion.

Comment (drakshakhan) [reply cmd/cluster/context_test.go]: Implemented the unit test cases using interfaces instead of mocking library as per the suggestion.

File: cmd/cluster/context_test.go [hunk 2]
```diff
@@ -0,0 +1,902 @@
...
+	if err == nil || err.Error() != "unknown Output Format: invalidOutputFormat" {
+		t.Errorf("Expected unknown output format error, got: %v", err)
+	}
+}
+
+func TestPrintUserBannedStatus(t *testing.T) {
+	tests := []struct {
+		name           string
+		data           contextData
+		expectedOutput string
```
[the remark this answers is not in the document]
Comment (drakshakhan) [reply cmd/cluster/context_test.go:869]: That was a miss on my part. I have now updated the test to properly compare the actual output with the expected output. The test captures stdout using a helper function (captureOutput) and verifies that the printed output matches the expected result. If there is a mismatch, the test fails with a detailed error message.

File: cmd/cluster/context_test.go [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+	for _, expected := range expectedStrings {
+		assert.Contains(t, output, expected, "Output should contain expected text: %s", expected)
+	}
+}
+
+func captureOutput(f func(io.Writer)) string {
+	var buf bytes.Buffer
+	f(&buf) // Passing the buffer as an io.Writer
+	return buf.String()
+}
```
Comment (joshbranham) [inline cmd/cluster/context_test.go:368]: You don't _really_ need this intermediate step. You can just define your `io.Writer` as a `bytes.Buffer` in tests, pass it to functions, then make assertions against it later. This is just a bit more indirection.

Comment (drakshakhan) [reply cmd/cluster/context_test.go]: Thanks for the suggestion! I simplified the code by removing the intermediate `captureOutput` function and using `bytes.Buffer` directly in the tests as recommended.
